src/components/navigation-control.js

```diff
@@ -40,7 +44,9 @@ export default class NavigationControl extends Component {
   _updateViewport(opts) {
     const {viewport} = this.context;
     const mapState = new MapState(Object.assign({}, viewport, opts));
-    this.props.onChangeViewport(mapState.getViewportProps());
+    // TODO(deprecate): remove this check when `onChangeViewport` gets deprecated
+    const onViewportChange = this.props.onChangeViewport || this.props.onViewportChange;
```

Comment:
I wanted to make it as easy as possible to move ahead with the duplication later on (i.e. in this case all you'd have to do is delete half of the conditional here) vs. having to also get rid of an import and change code.